### PR TITLE
Add vimeo/psalm

### LINKED ISSUE #73: Psalm integration
### Feature Request

|    Q        |   A
|------------ | ------
| QA    | yes

#### Summary

As decided during the [Technical-Steering-Committee Meeting on August 3rd, 2020](https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2020-08-03-TSC-Minutes.md#vote-on-static-analysis-tooling), Laminas wants to implement [vimeo/psalm](https://github.com/vimeo/psalm) in all packages.

Implementing psalm is quite easy.

#### Required
- [ ] Create a `.psalm.xml.dist` in the project root
- [ ] Copy and paste the contents from this [psalm.xml.dist](https://github.com/mezzio/repo-template/blob/6124090610ef1c6d6b0c80a063cafacc3ec68f38/psalm.xml.dist)
- [ ] Run `$ composer require vimeo/psalm`
- [ ] Run `$ vendor/bin/psalm --set-baseline=psalm-baseline.xml`
- [ ] Add a composer script `static-analysis` with the command `psalm --shepherd --stats`
- [ ] Add a new line to `script:` in `.travis.yml`: `- if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi`
- [ ] Remove phpstan from the project (`phpstan.neon.dist`, `.travis.yml` entry, `composer.json` `require-dev` and `scripts`)

##### Optional
- [ ] Fix as many psalm errors as possible.

### PR DESCRIPTION
**Pull request by lukaslangen**
Solves issue #73 

- [x] Create a .psalm.xml.dist in the project root
- [x] Copy and paste the contents from this psalm.xml.dist
- [x] Run $ composer require vimeo/psalm
- [x] Run $ vendor/bin/psalm --set-baseline=psalm-baseline.xml
- [x] Add a composer script static-analysis with the command psalm --shepherd --stats
- [x] Add a new line to script: in .travis.yml: - if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi
- [x] **(obsolete, because no phpstan configuration was given)** _Remove phpstan from the project (phpstan.neon.dist, .travis.yml entry, composer.json require-dev and scripts)_

### CODE REVIEW

**Comment by boesing** [comment]
@lukaslangen Thanks for this PR. 

To make travis pass, you can either temporarily add `COMPOSER_ROOT_VERSION=2.14.0` to `travis.yml` or `"version": "2.14.0"` to `composer.json` in a single commit.
After this, travis should startup and execute tests.
When anything is green, we can add this to the next release 👍

**Comment by weierophinney** [comment]
@lukaslangen Thank you!

I've rebased and pushed back to your branch. Since the 2.14.x series has a number of other changes, I'm resetting the baseline, and correcting a number of the autogenerated `@psalm-return` annotations on data providers; once those are done, I'll push back here, and it should be ready to merge.

Thanks for getting this started!

File: src/Barcode.php [hunk 1]
```diff
@@ -118,18 +118,19 @@ public function setAdapter($adapter, $options = null)
     /**
      * Returns the checksum option
      *
-     * @return string
+     * @return bool
      */
-    public function getChecksum()
+    public function getChecksum(): bool
```
Comment (Ocramius) [inline src/Barcode.php]: Thos change seems wrong 🤔

Comment (weierophinney) [reply src/Barcode.php]: Yeah, ran across another in the Bitwise validator. The original patch was on an older version of Psalm, so when I did my autofixes, it wasn't taking into account BC rules (which the v4 series does). Doing an audit of signatures now.

**Pull request merged**